### ISSUE TITLE
Argument 1 passed to Twig_Environment::addExtension() must implement interface Twig_ExtensionInterface, boolean given

### ISSUE THREAD

**Issue by dmitryChernukh** [symfony/symfony#28702]
**Symfony version(v3.4.16) affected**: 

**Description**  
The error occurred when we try to add twig extension:
```php
  $instance->addExtension(${($_ = isset($this->services['li_cms.twig.cms_extension']) ? $this->services['li_cms.twig.cms_extension'] : $this->getLiCms_Twig_CmsExtensionService()) && false ?: '_'});
```
If to be more precise - method: ```  $this->getLiCms_Twig_CmsExtensionService() ```

This method should return the instance of Twig_ExtensionInterface but boolean was returned. 
We found difference in method **-getLiCms_Twig_CmsExtensionService-**  between symfony version - 
3.4.15 and 3.4.16:

In version 3.4.15 
```php
protected function getLiCms_Twig_CmsExtensionService()
    {
        $a = ${($_ = isset($this->services['li_cms.service.cms_service']) ? $this->services['li_cms.service.cms_service'] : $this->getLiCms_Service_CmsServiceService()) && false ?: '_'};

        if (isset($this->services['li_cms.twig.cms_extension'])) {
            return $this->services['li_cms.twig.cms_extension'];
        }

        return $this->services['li_cms.twig.cms_extension'] = new \LI\CmsBundle\Twig\CmsExtension($a);
    }
```
and in version 3.4.16
```php
    protected function getLiCms_Twig_CmsExtensionService()
    {
        $a = ${($_ = isset($this->services['li_cms.service.cms_service']) ? $this->services['li_cms.service.cms_service'] : $this->getLiCms_Service_CmsServiceService()) && false ?: '_'};


        if (isset($this->privates['li_cms.twig.cms_extension'])) {
            return $this->privates['li_cms.twig.cms_extension'];
        }

        return $this->services['li_cms.twig.cms_extension'] = new \LI\CmsBundle\Twig\CmsExtension($a);
    }
```

as you can see in first example we take `li_cms.twig.cms_extension` from **$this->services** array and the second example - we take from **$this->privates**. 

The problem is:

**li_cms.twig.cms_extension** - in $this->privates array exists and it will return true:
![image](https://user-images.githubusercontent.com/16205596/46408202-03b3eb00-c71a-11e8-9df0-5be7b1eef8a4.png)

so this condition will be met:
```php
 if (isset($this->privates['li_cms.twig.cms_extension'])) {
            return $this->privates['li_cms.twig.cms_extension'];
        }
```
in this case function `getLiCms_Twig_CmsExtensionService()` - will return true instead of necessary class.

**Comment by nicolas-grekas** [comment]
Correct. Fixed by #28672 already. Thanks for the report.

**Comment by dmitryChernukh** [comment]
Thank you for feedback
